src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -450,7 +450,13 @@ private unsafe void CloseRequestQueueHandle()
             if ((_requestQueueHandle != null) && (!_requestQueueHandle.IsInvalid))
             {
                 if (NetEventSource.IsEnabled) NetEventSource.Info($"Dispose ThreadPoolBoundHandle: {_requestQueueBoundHandle}");
-                _requestQueueBoundHandle?.Dispose();
+
+                if (_requestQueueBoundHandle != null)
+                {
+                    _requestQueueBoundHandle.Dispose();
```

Comment:
Is it possible _requestQueueBoundHandle could end up getting set to null concurrently?  The previous implementation allowed for that, whereas this one could null ref.
```C#
ThreadPoolBoundHandle handle = _requestQueueBoundHandle;
if (handle != null)
{
    handle.Dispose();
    _requestQueueBoundHandle = null;
}
```

Reply:
No. `_requestQueueBoundHandle` is only ever used under `lock(_internalLock)`. We can put some `Assert(Monitor.IsEntered)` if we want -- that's already part of the larger change I started. This class unfortunately has several race conditions in it, so I didn't want to open the floodgades toward cleaning that stuff up in this PR.

Reply:
Ok